plugins/snapshot/init.go

```diff
@@ -0,0 +1,92 @@
+package snapshot
+
+import (
+	"github.com/gluster/glusterd2/servers/rest/route"
+	"github.com/prashanthpai/sunrpc"
+)
+
+// Plugin is a structure which implements GlusterdPlugin interface
+type Plugin struct {
+}
+
+// Name returns name of plugin
+func (p *Plugin) Name() string {
+	return "snapshot"
+}
+
+// SunRPCProgram returns sunrpc program to register with Glusterd
+func (p *Plugin) SunRPCProgram() sunrpc.Program {
+	return nil
+}
+
+// RestRoutes returns list of REST API routes to register with Glusterd
+func (p *Plugin) RestRoutes() route.Routes {
+	return route.Routes{
+		route.Route{
+			Name:        "SnapshotCreate",
+			Method:      "PUT",
+			Pattern:     "/snapshot/:snapname",
+			Version:     1,
+			HandlerFunc: snapshotCreateHandler},
+		route.Route{
+			Name:        "SnapshotActivate",
+			Method:      "POST",
+			Pattern:     "/snapshot/:snapname/activate",
+			Version:     1,
+			HandlerFunc: snapshotActivateHandler},
+		route.Route{
+			Name:        "SnapshotDeactivate",
+			Method:      "POST",
+			Pattern:     "/snapshot/:snapname/deactivate",
+			Version:     1,
+			HandlerFunc: snapshotDeactivateHandler},
+		route.Route{
+			Name:        "SnapshotClone",
+			Method:      "POST",
+			Pattern:     "/snapshot/:snapname/clone",
```

Comment:
let's take the code in , we can do an incremental update if required to the api.